Mathlib/Order/ScottContinuity/Prod.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2025 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import Mathlib.Order.ScottContinuity
+import Mathlib.Order.Bounds.Lattice
+
+/-!
+# Scott continuity on product spaces
+
+## Main result
+
+- `ScottContinuous_prod_of_ScottContinuous`: A function is Scott continuous on a product space if it
+  is Scott continuous in each variable.
+- `ScottContinuousOn.inf₂`: For complete linear orders, the meet operation is Scott continuous.
+
+-/
+
+open Set
+
+variable {α β γ : Type*}
+
+/- `f` is Scott continuous on a product space if it is Scott continuous and monotone in each
+variable -/
+lemma ScottContinuousOn_prod_of_ScottContinuousOn [Preorder α] [Preorder β] [Preorder γ]
+    {f : α × β → γ} {D : Set (Set (α × β))}
+    (h₁ : ∀ a, ScottContinuousOn ((fun d => Prod.snd '' d) '' D) (fun b => f (a,b)))
+    (h₂ : ∀ b, ScottContinuousOn ((fun d => Prod.fst '' d) '' D) (fun a => f (a,b)))
+    (h₁' : ∀ a, Monotone (fun b => f (a,b))) (h₂' : ∀ b, Monotone (fun a => f (a,b))) :
+    ScottContinuousOn D f := fun d hX hd₁ hd₂ ⟨p1, p2⟩ hdp => by
+  rw [isLUB_congr ((monotone_prod_iff.mpr ⟨(fun a => (h₁' a)),
+    ((fun b => (h₂' b)))⟩).upperBounds_image_of_directedOn_prod hd₂),
+    ← iUnion_of_singleton_coe (Prod.fst '' d), iUnion_prod_const, image_iUnion,
+    ← isLUB_iUnion_iff_of_isLUB (fun a => by
+      rw [singleton_prod, image_image f (fun b ↦ (a, b))]
+      exact h₁ _ (mem_image_of_mem (fun d ↦ Prod.snd '' d) hX) (Nonempty.image Prod.snd hd₁)
+        (DirectedOn.snd hd₂) ((isLUB_prod (_,_)).mp hdp).2) _, Set.range]
+  convert (h₂ _
+    (mem_image_of_mem (fun d ↦ Prod.fst '' d) hX) (Nonempty.image Prod.fst hd₁) (DirectedOn.fst hd₂)
+    ((isLUB_prod (p1,p2)).mp hdp).1)
+  ext : 1
+  simp_all only [Prod.forall, Prod.mk_le_mk, and_imp, Subtype.exists, mem_image, Prod.exists,
+    exists_and_right, exists_eq_right, exists_prop, mem_setOf_eq]
+
+lemma ScottContinuous_prod_of_ScottContinuous {γ : Type*} [Preorder α] [Preorder β] [Preorder γ]
```

Comment:
```suggestion
lemma ScottContinuous.fromProd {γ : Type*} [Preorder α] [Preorder β] [Preorder γ]
```
maybe?? Can you also add `ScottContinuous.prod (hf : ScottContinuous f) (hg : ScottContinuous g) : ScottContinuous (Prod.map f g)`?